### PR TITLE
Add instruments for gorilla/mux

### PR DESCRIPTION
**Pull request by izaurio**
* RouteName helper

### CODE REVIEW

File: nrgorilla/nrgorilla_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package nrgorilla_test
```
Comment (mkabischev) [inline nrgorilla/nrgorilla_test.go:1]: Really no reason to use `nrgorilla_test` instead of `nrgorilla`.

File: nrgorilla/nrgorilla_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"testing"
```
Comment (mkabischev) [inline nrgorilla/nrgorilla_test.go]: please sort imports

File: nrgorilla/nrgorilla.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// InstrumentRoutes adds instrumentation to a router.
+// This must be used after the routes have been added to the router.
+func InstrumentRoutes(r *mux.Router, app newrelic.Application) *mux.Router {
+	r.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error {
+		h := newrelicutil.WrapHandler(app, RouteName(route), route.GetHandler())
+		route.Handler(h)
+
+		return nil
+	})
+	if nil != r.NotFoundHandler {
```
Comment (mkabischev) [inline nrgorilla/nrgorilla.go]: Why do you prefer such style?
```
if .r.NotFoundHandler != nil {
    r.NotFoundHandler = newrelicutil.WrapHandler(app, "NotFoundHandler", r.NotFoundHandler)
}
```

File: nrgorilla/nrgorilla.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// Package nrgorilla provides New Relic instruments for gorilla/mux tool.
+package nrgorilla
+
+import (
+	"strings"
+	"github.com/gorilla/mux"
```
Comment (mkabischev) [inline nrgorilla/nrgorilla.go]: blank line required after `strings`.

**Review by mkabischev — CHANGES_REQUESTED** [review]

File: nrgorilla/nrgorilla.go (hunk 3 above)
Comment (izaurio) [reply nrgorilla/nrgorilla.go]: Copy paste from code base with another code style

File: nrgorilla/nrgorilla_test.go (hunk 1 above)
Comment (izaurio) [reply nrgorilla/nrgorilla_test.go:1]: there's no reason not to do it!

Comment (mkabischev) [reply nrgorilla/nrgorilla_test.go:1]: In Go by default people use the same package name. The main reason to use `*_test` package when you cycle dependencies.

**Pull request merged**